### ISSUE TITLE
Invalid Reference(#name)

### ISSUE THREAD

**Issue by markkramerus** [FHIR/GoFSH#120]
This example: [http://hl7.org/fhir/us/core/2021Jan/MedicationRequest-uscore-mo2.json](http://hl7.org/fhir/us/core/2021Jan/MedicationRequest-uscore-mo2.json)
has a contained instance of a medication `med2`.  This line in the generated FSH looks wrong:
```
* medicationReference.reference = "#med2"
```
because it has an extra # sign. Clearly this is not a code. Is this a bug?

**Comment by cmoesel** [comment]
That's technically correct because FHIR references contained resources as fragments, using `#`.  This matches what's in the source you linked to:
```json
  "medicationReference": {
    "reference": "#med2",
    "display": "Nizatidine 15 MG/ML Oral Solution [Axid]"
  }
```
If it were a code, it would not be wrapped in `""` (e.g., `"#med2"` vs `#med2`).

That said, I think a best practice FSH representation would be:
```
* medicationReference = Reference(med2) "Nizatidine 15 MG/ML Oral Solution [Axid]"
```
But what GoFSH did is not _wrong_.

**Comment by cmoesel** [comment]
Actually, I found that if I ran GoFSH on the linked JSON and specified US Core as a dependency when running GoFSH, it _did_ result in a real bug:
```
* medicationReference = Reference(#med2) "Nizatidine 15 MG/ML Oral Solution [Axid]"
```
Not sure why the US Core dependency made a difference (I guess something resolved that didn't resolve before).  While the above is closer to the best practice approach, in this case `#med2` is not wrapped in `""`, so it is a bug.  It needs to either be `Reference("#med2")` or `Reference(med2)`, but definitely NOT `Reference(#med2)`.
